packages/passport-interface/src/SubscriptionManager.ts

```diff
@@ -0,0 +1,244 @@
+import { Emitter } from "@pcd/emitter";
+import { PCDAction, PCDPermission } from "@pcd/pcd-collection";
+import {
+  ArgsOf,
+  PCDOf,
+  PCDPackage,
+  PCDTypeNameOf,
+  SerializedPCD
+} from "@pcd/pcd-types";
+import { IFeedApi } from "./FeedAPI";
+
+/**
+ * Class responsible for storing the list of feed providers this application is
+ * aware of, as well as the list of feeds that each provider can serve, and which
+ * of those we are subscribed to.
+ */
+export class FeedSubscriptionManager {
+  public updatedEmitter: Emitter;
+  private api: IFeedApi;
+  private providers: SubscriptionProvider[];
+  private activeSubscriptions: Subscription[];
+
+  public constructor(
+    api: IFeedApi,
+    providers?: SubscriptionProvider[],
+    activeSubscriptions?: Subscription[]
+  ) {
+    this.updatedEmitter = new Emitter();
+    this.api = api;
+    this.providers = providers ?? [];
+    this.activeSubscriptions = activeSubscriptions ?? [];
+  }
+
+  public async listFeeds(providerUrl: string): Promise<Feed[]> {
+    return this.api.listFeeds(providerUrl).then((r) => r.feeds);
+  }
+
+  public async pollSubscriptions(): Promise<SubscriptionActions[]> {
+    const responses: SubscriptionActions[] = [];
+
+    for (const subscription of this.activeSubscriptions) {
+      try {
+        responses.push({
+          actions: (
+            await this.api.pollFeed(subscription.providerUrl, {
+              feedId: subscription.feed.id,
+              pcd: subscription.credential
+            })
+          ).actions,
+          subscription
+        });
+      } catch (e) {
+        console.log(`failed to poll subscription`, e);
```

Comment:
i imagine over time some subscriptions will go stale and always fail - how would we expose that to the user?

Reply:
This should probably appear on the subscription management screen, perhaps with a red warning icon next to the feed name or something similar. I haven't implemented any new UI in this PR, but I definitely want to revisit this.